packages/databricks-vscode/src/file-managers/ProjectConfigFile.test.ts

```diff
@@ -1,48 +1,140 @@
 
 import {mkdir, mkdtemp, readFile, writeFile} from "fs/promises";
-import {ProjectConfig, ProjectConfigFile} from "./ProjectConfigFile";
+import {ProjectConfigFile} from "./ProjectConfigFile";
 import * as assert from "assert";
 import path from "path";
 import * as os from "os";
 import {ProfileAuthProvider} from "../configuration/auth/AuthProvider";
+import {LocalUri, RemoteUri} from "../sync/SyncDestination";
+import {WorkspaceStateManager} from "../vscode-objs/WorkspaceState";
+import {DatabricksYamlFile} from "./DatabricksYamlFile";
+import {ProjectJsonFile} from "./ProjectJsonFile";
+import * as YAML from "yaml";
 import {Uri} from "vscode";
 
 describe(__filename, () => {
     let tempDir: string;
+    let mockWorkspaceStateManager: WorkspaceStateManager;
     before(async () => {
         tempDir = await mkdtemp(path.join(os.tmpdir(), "ProjectConfTests-"));
+        const fixedUUID = "uuid1234-something";
+        mockWorkspaceStateManager = {
+            fixedUUID,
+        } as WorkspaceStateManager;
     });
 
-    it("should write config file", async () => {
+    it("should write project.json and databricks.yaml config files", async () => {
         const authProvider = new ProfileAuthProvider(
             new URL("https://000000000000.00.azuredatabricks.net/"),
             "testProfile"
         );
-        const expected: ProjectConfig = {
-            authProvider: authProvider,
-            clusterId: "testClusterId",
-            workspacePath: Uri.from({scheme: "wsfs", path: "workspacePath"}),
-        };
-        await new ProjectConfigFile(expected, tempDir, "databricks").write();
+        await new ProjectConfigFile(
+            new ProjectJsonFile(authProvider),
+            new DatabricksYamlFile(
+                authProvider.host,
+                "testClusterId",
+                new RemoteUri("workspacePath")
+            )
+        ).write(new LocalUri(tempDir), mockWorkspaceStateManager);
 
-        const rawData = await readFile(
-            ProjectConfigFile.getProjectConfigFilePath(tempDir),
-            {encoding: "utf-8"}
+        const packageJsonData = JSON.parse(
+            await readFile(
+                ProjectJsonFile.getFilePath(new LocalUri(tempDir)).path,
+                {encoding: "utf-8"}
+            )
         );
-        const actual = JSON.parse(rawData);
-        assert.deepEqual(actual, {
+
+        assert.deepEqual(packageJsonData, {
             host: "https://000000000000.00.azuredatabricks.net/",
             authType: "profile",
             profile: "testProfile",
-            workspacePath: "workspacePath",
-            clusterId: "testClusterId",
         });
+
+        const yamlData = YAML.parse(
+            await readFile(
+                DatabricksYamlFile.getFilePath(new LocalUri(tempDir)).path,
+                {encoding: "utf-8"}
+            )
+        );
+
+        assert.deepEqual(yamlData, {
+            environments: {
```

Comment:
Why are we using environments? Shouldn't we just overwrite the top-level keys?

Reply:
Top level keys might be present in a user defined bundle.yaml. The only way to safely override them (while allowing users the ability to target other envs from terminal) is to use a vscode specific env to override the configs. This is the only env users can edit through the UI and is never committed to git.

Reply:
but it also means that users can't easily use it from the command line. Unless we make the VSCode env the default.